lib/plugins/cleaner.rb

```diff
@@ -0,0 +1,51 @@
+require 'rubygems/package'
+require_relative '../moonshot/creds_helper'
+
+module Moonshot
+  module Plugins
+    # Moonshot plugin class for removing artifacts when a stage gets deleted.
+    class Cleaner
+      include Moonshot::CredsHelper
+
+      def initialize(bucket_name)
+        @bucket_name = bucket_name
+      end
+
+      def post_delete(resources)
+        stack_name = resources.stack.name
+        resources.ilog.start 'Starting to delete artifacts for stack: ' \
+          "#{stack_name}" do |s|
+          arts = retrieve_artifact_names(stack_name)
+          if arts.empty?
+            s.success 'No artifacts found, nothing deleted.'
+          else
+            delete_artifacts arts
+            s.success "Deleted artifacts for stack: '#{stack_name}' " \
+                      "in bucket: '#{@bucket_name}'"
+          end
+        end
+      end
+
+      private
+
+      def retrieve_artifact_names(name)
+        objs = s3_client.list_objects(bucket: @bucket_name, prefix: name)
+        valid_keys = []
+        objs.contents.each do |o|
+          valid_keys << o.key if o.key =~ /^#{name}[\.-]/
```

Comment:
`\d` is not enough. Use of the above example from @askreet : `dev-chris` and `dev-chris-test1` and `dev-chris-123412134`

How can a filter determine if test1 is a different name or just a version number for `dev-chris`? If I define a fixed version (named)  as `test` for my `dev-yitsushi` stage it will be `dev-yitsushi-test.tar.gz` but if I create a stage as `dev-yitsushi-test` it will be `dev-yitsushi-test-test.tar.gz`

Btw it can be a problem in general. There is 2 stage with a shared S3 bucket:
 - `mysite`
 - `mysite-test`

I create a version for `mysite-test` with a single digit `1` and create a `test-1` version for `mysite` => both will be `mysite-test-1`.